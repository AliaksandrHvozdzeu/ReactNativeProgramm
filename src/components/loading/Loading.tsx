import React from 'react';
import {styles} from './styles';
import {ActivityIndicator, Text, View} from 'react-native';
import {COLORS} from '../../utils/colors';

type LoadingProps = {};
const Loading: React.FC<LoadingProps> = () => {
  return (
    <View style={styles.onLoadDataContainer}>
      <ActivityIndicator size="large" color={COLORS.blue_500} />
      <View>
        <Text style={styles.loadingData}>Loading...</Text>
      </View>
    </View>
  );
};

export default Loading;
