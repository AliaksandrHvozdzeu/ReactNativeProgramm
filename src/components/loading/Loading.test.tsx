import 'react-native';
import React from 'react';
import {test, describe, expect} from '@jest/globals';
import {render} from '@testing-library/react-native';
import Loading from './Loading';
import {Text} from 'react-native';

describe('Loading tests', () => {
  test('Renders the Loading correctly', () => {
    const tree = render(<Loading />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Loading render works correctly', () => {
    render(<Loading />);
  });

  test('Component Text presented in component', () => {
    const wrapper = render(<Loading />);
    const testInstance = wrapper.root;
    expect(testInstance.findByType(Text)).toBeTruthy();
  });

  test('Component Text is not empty', () => {
    const wrapper = render(<Loading />);
    const testInstance = wrapper.root;
    const textValue = testInstance.findByType(Text);
    expect(textValue.props.children).toBe('Loading...');
  });
});
