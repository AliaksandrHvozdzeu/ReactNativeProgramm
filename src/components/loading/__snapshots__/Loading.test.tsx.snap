// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading tests Renders the Loading correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "height": 600,
      "justifyContent": "center",
      "padding": 10,
      "verticalAlign": "middle",
    }
  }
>
  <ActivityIndicator
    color="#008ACE"
    size="large"
  />
  <View>
    <Text
      style={
        {
          "borderColor": "#8F8F8F",
          "display": "flex",
          "fontFamily": "Roboto",
          "fontSize": 15,
          "fontStyle": "normal",
          "fontWeight": "700",
          "left": 5,
          "marginLeft": 10,
        }
      }
    >
      Loading...
    </Text>
  </View>
</View>
`;
